modules/vault/src/main/java/org/apache/ignite/internal/vault/persistence/PersistentVaultService.java

```diff
@@ -94,59 +92,70 @@ public void start() {
                                 .setFilterPolicy(new BloomFilter(10, false))
                                 .setOptimizeFiltersForMemory(true)
                 );
-
-        try {
-            db = RocksDB.open(options, path.toString());
-        } catch (RocksDBException e) {
-            throw new IgniteInternalException(e);
-        }
     }
 
-    /** {@inheritDoc} */
     @Override
-    public void stop() {
-        // TODO: IGNITE-15161 Implement component's stop.
+    public void start() {
         try {
-            close();
+            db = RocksDB.open(options, path.toString());
         } catch (RocksDBException e) {
             throw new IgniteInternalException(e);
         }
     }
 
     /** {@inheritDoc} */
     @Override
-    public void close() throws RocksDBException {
-        db.syncWal();
-
+    public void close() throws Exception {
         IgniteUtils.shutdownAndAwaitTermination(threadPool, 10, TimeUnit.SECONDS);
```

Comment:
`InFlightFutures` can be used to track and, on stop, cancel the futures that will never be resolved due to not having a chance to be executed